qiskit/transpiler/passes/layout/sabre_layout.py

```diff
@@ -69,6 +71,18 @@ class SabreLayout(TransformationPass):
     layout pass. When specified this will use the specified routing pass to select an
     initial layout only and will not run multiple seed trials.
 
+    In addition to starting with a random initial `Layout` the pass can also take in
+    an additional list of starting layouts which will be used for additional
+    trials. If the ``sabre_starting_layout`` is present in the property set
+    when this pass is run, that will be used for additional trials. There will still
+    be ``layout_trials`` of full random starting layouts run and the contents of
+    ``sabre_starting_layout`` will be run in addition to those. The output which results
+    in the lowest amount of swap gates (whether from the random trials or the property
+    set starting point) will be used. The value for this property set field should be a
+    list of :class:`.Layout` objects representing the starting layouts to use. If a
+    virtual qubit is missing from an :class:`.Layout` object in the list a random qubit
+    will be selected.
+
```

Comment:
Related to the other comment in this file: since we're doing more of this now, I think it might be good to have a clearer way to document which property-set variables affect a pass's operation.  Something like
```rst
Properties Read
---------------

``sabre_starting_layout`` (``list[Layout]``)
    In addition to starting with a random initial :class:`.Layout`, ...

Properties Written
------------------
``layout`` (:class:`.Layout`)
    The chosen initial mapping of virtual to physical qubits, including the ancilla allocation.

``final_layout`` (:class:`.Layout`)
    A permutation of how swaps have been applied to the input qubits at the end of the circuit. 
```
(or whatever descriptions).

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/10721/commits/ebd74ad1cdd8ece706cb42269a2bac2340d04c13